### PR TITLE
feat: Integrate kernel with bundleStore

### LINKED ISSUE #7197: integrate bundleStore into kernel
## What is the Problem Being Solved?

To finish off #7089, we need the kernel to actually use the new `bundleStore`.

## Description of the Design

* `kernelKeeper.addBundle` changes to use `bundleStore.addBundle` instead of stringifying the bundle and adding it to `kvStore`
* `hasBundle` and `getBundle` change to match
* `(add|get)NamedBundleID` remain the same: the name-to-bundleID mapping remains in kvStore
* some tests probably need updating

## Security Considerations

None, same as listed in #7089

## Scaling Considerations

same as #7089 

## Test Plan

don't really need any

### PR DESCRIPTION
**Pull request by FUDCo**
Closes #7197

### CODE REVIEW

File: packages/SwingSet/src/kernel/state/kernelKeeper.js [hunk 1]
```diff
@@ -432,34 +421,31 @@ export default function makeKernelKeeper(kernelStorage, kernelSlog) {
...
    * @param {BundleID} bundleID
    * @returns {EndoZipBase64Bundle | undefined}
    */
   function getBundle(bundleID) {
-    const value = kvStore.get(bundleIDToKey(bundleID));
-    if (value) {
-      return JSON.parse(value);
+    if (bundleStore.hasBundle(bundleID)) {
+      const bundle = bundleStore.getBundle(bundleID);
+      if (bundle.moduleFormat !== 'endoZipBase64')
```
Comment (warner) [inline packages/SwingSet/src/kernel/state/kernelKeeper.js]: you ok with the lack of braces? I am, apparently :)

File: packages/SwingSet/test/bundling/test-bundles-kernel.js [hunk 2]
```diff
@@ -46,25 +40,26 @@ test('install bundle', async t => {
...
   await kernel.installBundle(bundleID, bundle);
-  t.deepEqual(JSON.parse(kvStore.get(`bundle.${bundleID}`)), bundle);
+  t.deepEqual(bundleStore.getBundle(bundleID), bundle);
 
   // the kernel's installBundle() doesn't validate
   const wrong =
     'b1-00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000';
-  await kernel.installBundle(wrong, bundle);
-  t.deepEqual(JSON.parse(kvStore.get(`bundle.${wrong}`)), bundle);
+  await t.throwsAsync(() => kernel.installBundle(wrong, bundle), {
```
Comment (warner) [inline packages/SwingSet/test/bundling/test-bundles-kernel.js]: Huh, this is at odds with the "the kernel's installBundle() doesn't validate" comment above. I'm happy to have `kernel.installBundle()` do validation, but we should make sure we understand where the validation is taking place. Then let's change the comment.

File: packages/swing-store/src/bundleStore.js [hunk 3]
```diff
@@ -72,6 +72,7 @@ export function makeBundleStore(db, ensureTxn, noteExport = () => {}) {
    * @param {Bundle} bundle
    */
   function addBundle(bundleID, bundle) {
+    bundleID.startsWith('b1-') || Fail`invalid bundleID ${q(bundleID)}`;
```
Comment (warner) [inline packages/swing-store/src/bundleStore.js:75]: This will collide with my #7242, whichever one lands last will need to resolve a merge conflict.

**Review by warner — APPROVED** [review]
small changes to make, then feel free to land. Be aware of likely conflict with #7242.

File: packages/SwingSet/src/kernel/state/kernelKeeper.js (hunk 1 above)
Comment (FUDCo) [reply packages/SwingSet/src/kernel/state/kernelKeeper.js]: ¡Doh!

File: packages/SwingSet/test/bundling/test-bundles-kernel.js (hunk 2 above)
Comment (FUDCo) [reply packages/SwingSet/test/bundling/test-bundles-kernel.js]: Yeah, this test fragment is now redundant with the "too short" check a few lines above it.  My inclination is to just take it out.

File: packages/swing-store/src/bundleStore.js (hunk 3 above)
Comment (FUDCo) [reply packages/swing-store/src/bundleStore.js:75]: I think we're both very close to landing, so the question is probably which order will make the conflict resolution easiest.

**Pull request merged**